app/routes/goal_routes.py

```diff
@@ -0,0 +1,59 @@
+from flask import Blueprint, make_response, request
+from app.models.goal import Goal
+from app.models.task import Task
+from app.routes.helpers import validate_model, create_item, get_all_items, get_one_item, update_item, delete_item
```

Comment:
This is an interesting approach! Abstracting away all the logic for the routes into a helper file definitely makes all of your goal routes very concise. It works here since goals and tasks are pretty similar overall. In the future, you may find that there isn't as much similarity between models and you'll need to write your logic in each of your routes.